Code/Framework/AzCore/AzCore/Serialization/Json/JsonImporter.cpp

```diff
@@ -144,21 +160,606 @@ namespace AZ
+        importPathStack.pop_back();
+
+        if (result.GetOutcome() == JsonSerializationResult::Outcomes::Catastrophic)
+        {
+            return result;
+        }
+
+        return JsonSerializationResult::ResultCode(JsonSerializationResult::Tasks::Import,
+            JsonSerializationResult::Outcomes::Success);
+    }
+
+    auto JsonImportResolver::GetImportPaths(const rapidjson::Value& importDirective,
+        const ImportPathStack& importPathStack)
+        -> ImportPathsResult
+    {
+        AZ::IO::FixedMaxPath importAbsPath = importPathStack.back();
+        importAbsPath.RemoveFilename();
+        AZ::IO::FixedMaxPath importName;
+        if (importDirective.IsObject())
+        {
+            auto filenameField = importDirective.FindMember("filename");
+            if (filenameField != importDirective.MemberEnd())
+            {
+                importName = AZ::IO::FixedMaxPath(AZStd::string_view(
+                    filenameField->value.GetString(), filenameField->value.GetStringLength()));
+            }
+        }
+        else
+        {
+            importName = AZ::IO::FixedMaxPath(AZStd::string_view(
+                importDirective.GetString(), importDirective.GetStringLength()));
+        }
+
+        // Resolve the any file @..@ aliases in the relative importName if it starts with one
+        if (auto fileIo = AZ::IO::FileIOBase::GetInstance(); fileIo != nullptr)
+        {
+            // Replace alias doesn't "resolve" the path as FileIOBase::ResolvePath would
+            // It only replaces an alias, it doesn't make a relative path absolute by
+            // making subpath of the asset cache
+            fileIo->ReplaceAlias(importName, importName);
+        }
+
+        importAbsPath /= importName;
+
+        ImportPathsResult pathsResult;
+        pathsResult.m_importAbsPath = importAbsPath;
+        pathsResult.m_importRelPath = importName;
+
+        return pathsResult;
+    }
+
+    // The algorithm below is a modified JSON Merge Patch algorithm that is for patching a specific field
+    // in the target JSON value
+    // The modifications are that any `null` entries do not revome existing fields from the target JSON
+    // The reason this is needed is because the Json Import operation should preserve existing fields
+    // in case the JSON data is used for an JSON patch or JSON merge patch operation explicitly
+    JsonSerializationResult::ResultCode JsonImportResolver::PatchField(rapidjson::Value& target,
+        rapidjson::Document::AllocatorType& allocator, const rapidjson::Value& source,
+        rapidjson::Value::StringRefType fieldNameStringRef, const rapidjson::Value& fieldValue,
+        ImportPathStack& importPathStack,
+        JsonImportSettings& settings, StackedString& element)
+    {
+        using namespace JsonSerializationResult;
+
+        using ReporterString = AZStd::fixed_string<1024>;
```

Comment:
Should this definition get pushed out somewhere a little more global, so that we don't have a bunch of hard-coded fixed_string<1024> entries everywhere?

Reply:
Yes, I'll add this to the [JsonSerializationResult.h](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Serialization/Json/JsonSerializationResult.h#L117) header since the primary purpose of the typedef is to construct a temporary string without dynamic allocations for use in the JsonIssueCallback function to pass in the error message.